File: packages/shared/src/types/application.ts
export interface Application {
  id: number;
  job_id: number;
  resume_path: string;
  cover_letter?: string;
  notes?: string;
  status: ApplicationStatus;
  submitted_at?: Date;
  created_at: Date;
  updated_at: Date;
}

export interface ApplicationCreate {
  job_id: number;
  resume_path: string;
  cover_letter?: string;
  notes?: string;
}

export enum ApplicationStatus {
  DRAFT = "draft",
  SUBMITTED = "submitted",
  UNDER_REVIEW = "under_review",
  INTERVIEW = "interview",
  REJECTED = "rejected",
  ACCEPTED = "accepted"
}

File: packages/backend/src/models/application.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from enum import Enum

class ApplicationStatus(str, Enum):
    DRAFT = "draft"
    SUBMITTED = "submitted"
    UNDER_REVIEW = "under_review"
    INTERVIEW = "interview"
    REJECTED = "rejected"
    ACCEPTED = "accepted"

class ApplicationBase(BaseModel):
    job_id: int
    resume_path: str
    cover_letter: Optional[str] = None
    notes: Optional[str] = None

class ApplicationCreate(ApplicationBase):
    pass

class ApplicationUpdate(BaseModel):
    resume_path: Optional[str] = None
    cover_letter: Optional[str] = None
    notes: Optional[str] = None
    status: Optional[ApplicationStatus] = None

class Application(ApplicationBase):
    id: int
    status: ApplicationStatus = ApplicationStatus.DRAFT
    submitted_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True